_posts/en/newsletters/2024-01-10-newsletter.md

```diff
@@ -0,0 +1,393 @@
+      validity, but it can't be used as an upper bound
+      because future block timestamps are unknown."
+  a3link="https://bitcoincore.reviews/28956#l-77"
+
+  q4="Why are limits enforced on how far \"off\" a block
+      header's timestamp is allowed to be from the node's internal clock?
+      And since we don't require exact agreement on time, can these
+      limits be made more strict?"
+  a4="The block timestamp range is restricted so that malicious
+      nodes' ability to manipulate difficulty adjustments and
+      locktimes is limited. These kinds of attacks are called
+      timewarp attacks.
+      The valid range can be made more strict to an extent,
+      but making it too strict could lead to temporary chain splits
+      since some nodes may reject blocks that other nodes accept.
+      Block timestamps don't need to be exactly correct, but
+      they need to track actual time over the long run."
+  a4link="https://bitcoincore.reviews/28956#l-82"
+
+  q5="Before this PR, why would an attacker try to manipulate
+      a node's network-adjusted time?"
+  a5="If the node is a miner, to get its mined blocks rejected
+      by the network or to get it to not accept a valid block so
+      it keeps wasting hashrate on an old tip
+      (both of those would give an advantage to a competing miner);
+      to get the attacked node to follow the wrong chain;
+      to cause a time-locked transaction to not be mined when
+      it should be;
+      to perform a [time dilation attack][] on the Lightning Network."
+  a5link="https://bitcoincore.reviews/28956#l-89"
+
+  q6="Prior to this PR, how could an attacker try to manipulate a
+      node's network-adjusted time? Which network message(s) would
+      they use?"
+  a6="An attacker would need to send us version messages with
+      manipulated timestamps from multiple peers that they control.
+      They would need us to make more than 50% of our outbound
+      connections to their nodes, which is hard but much easier
+      than completely eclipsing the node."
+  a6link="https://bitcoincore.reviews/28956#l-100"
+
+  q7="This PR uses the node's local clock as the upper-bound
+      block validation time, rather than network-adjusted time.
+      Can we be sure that this reduces esoteric attack surfaces,
+      rather than increasing them?"
+  a7="Discussion ensued with no clear resolution as to whether it's
+      easier for an attacker to affect a node's peer set or its
+      internal clock (using malware or NTP fakery, for example),
+      but most participants agreed that the PR is an improvement."
+  a7link="https://bitcoincore.reviews/28956#l-102"
+
+  q8="Does this PR change consensus behavior? If so, is this a
+      soft fork, a hard fork, or neither? Why?"
+  a8="It's not a consensus change, it's just a network acceptance
+      policy change. Even though block timestamp acceptance is
+      not part of consensus, it's [essential][se timestamp accecptance]
```

Comment:
@LarryRuane 

I think "block timestamp acceptance" is general enough a concept to include MTP, and MTP is clearly part of consensus.  Maybe this could be phrased a bit differently to clarify that there can be no consensus rule that includes data from outside the block chain, such as data from each user's own clock.

Reply:
Excellent point, pushed a fixup commit (1700347ea4dcc222c4ea1f4725c43c632c5a8381). Let me know if it could be improved further, thanks.